packages/quantic/force-app/main/default/lwc/quanticFeedbackModal/quanticFeedbackModal.js

```diff
@@ -0,0 +1,234 @@
+    explainWhy,
+    selectOneOfOptions,
+    selectTheReason,
+    whatWasTheReason,
+    fillOutField,
+    yourResponsesHelp,
+    thankYouForYourFeedback,
+  };
+
+  /** @type {string} */
+  feedbackValue;
+  /** @type {string} */
+  detailsValue;
+  /** @type {boolean} */
+  feedbackSubmitted = false;
+  /** @type {boolean} */
+  detailsInputError = false;
+  /** @type {string} */
+  error;
+  /** @type {boolean} */
+  validated = false;
+
+  connectedCallback() {
+    this.validateFeedbackModal();
+  }
+
+  /**
+   * Validates the properties provided to the feedback modal.
+   */
+  validateFeedbackModal() {
+    getBueno(this).then(() => {
+      this.validateTheOptionsProperty();
+      if (!this.error) {
+        this.validateOptions();
+      }
+      if (!this.error) {
+        this.validateTheSubmitFeedbackProperty();
+      }
+      this.validated = true;
+    });
+  }
+
+  validateTheOptionsProperty() {
+    if (!Bueno.isArray(this.options)) {
+      console.error('The options provided are not in a valid array.');
+      this.setError();
+    } else if (!this.options.length) {
+      console.error('At least one option must be specified.');
+      this.setError();
+    }
+  }
+
+  validateOptions() {
+    const missingValueOrLabel = this.options?.some(
+      ({value, label}) => !value || !label
+    );
+    if (missingValueOrLabel) {
+      console.error(
+        'Each option requires a label and a value to be specified.'
+      );
+      this.setError();
+    } else {
+      this.options.forEach(({value, label}) => {
+        if (!Bueno.isString(value)) {
+          console.error(`The "${value}" value is not a valid string.`);
+          this.setError();
+        }
+        if (!Bueno.isString(label)) {
+          console.error(`The "${label}" label is not a valid string.`);
+          this.setError();
+        }
+      });
+    }
+  }
+
+  validateTheSubmitFeedbackProperty() {
+    if (typeof this.submitFeedback !== 'function') {
+      console.error('The submitFeedback property is not a valid function.');
+      this.setError();
+    }
+  }
+
+  setError() {
+    this.error = 'c-quantic-feedback-modal Error';
+  }
+
+  /**
+   * Closes the modal.
+   * @returns {void}
+   */
+  closeModal() {
+    this.close();
+  }
+
+  /**
+   * Returns the lightning-radio-group element.
+   * @returns {{setCustomValidity: function, reportValidity: function}}
+   */
+  get optionsInput() {
+    return this.template.querySelector('lightning-radio-group');
+  }
+
+  /**
+   * Returns the CSS class of the reason input.
+   */
+  get detailsFormClass() {
+    return `slds-form-element ${
+      this.detailsInputError ? 'slds-has-error' : ''
+    }`;
+  }
+
+  /**
+   * Handles the option change.
+   * @param {CustomEvent} event
+   */
+  handleOptionChange(event) {
+    this.detailsInputError = false;
+    this.optionsInput.setCustomValidity();
+    this.optionsInput.reportValidity();
+    this.feedbackValue = event.detail.value;
```

Comment:
Why don't you store here the `this.options` that match the `event.detail.value` ?